common/models/user.js

```diff
@@ -360,20 +367,19 @@ module.exports = function(User) {
     // Email model
     var Email = options.mailer || this.constructor.email || loopback.getModelByType(loopback.Email);
 
-    crypto.randomBytes(64, function(err, buf) {
-      if (err) {
-        fn(err);
-      } else {
-        user.verificationToken = buf.toString('hex');
-        user.save(function(err) {
-          if (err) {
-            fn(err);
-          } else {
-            sendEmail(user);
-          }
-        });
-      }
-    });
+    // Set a default token generation function if one is nto provided
+    var tokenGenerator = options.generateVerificationToken || User.generateVerificationToken;
+
+    tokenGenerator.apply(user, [function(token) {
```

Comment:
Our default one doesn't, no. But the thought was to provide the custom token generator function that a developer might implement with a proper context within which to generate the token. For example, if the token needed to be a hash of the user's email address and some salt.